src/System.Management.Automation/engine/debugger/debugger.cs

```diff
@@ -2938,6 +3189,10 @@ private Debugger PopActiveDebugger()
                             _steppingMode = SteppingMode.StepIn;
                             _overOrOutFrame = _nestedRunningFrame;
                             _nestedRunningFrame = null;
+                            if (_lastActiveDebuggerAction == DebuggerResumeAction.StepOut)
+                            {
+                                CancelDebuggerProcessingAsNeeded(poppedDebugger);
```

Comment:
The CancelDebuggerProcessingAsNeeded() method cancels debugger processing only when stepping out of the current call stack, which means stepping out of the currently debugged remote session.  This behavior was debated but we finally decided to step out of all remote sessions from the Invoke-Command, and stop in the local script that called Invoke-Command.  

We decided to do this because stepping out of one remote session and into another results in confusion because you are debugging one remote session while one or more are running in parallel spewing output into your debug session console.  So "step-out" at the top of the stack means to step out of the Invoke-Command remote debugging and back to the local script.

I'll add a comment to try to make this clear.